tests/test_chat_to_files.py

```diff
@@ -1,6 +1,21 @@
 import textwrap
 
 from gpt_engineer.core.chat_to_files import to_files
+from gpt_engineer.core.db import DB, DBs
+
+from typing import Union
+from pathlib import Path
+
+
+class StubDB(DB):
```

Comment:
Id appreciate some feedback here. I put this code in because i was getting permissions errors when running the unit tests on the build agent. My solution was to mock the file system calls. However my gut feeling from reading around is that this isn't the 'python' way of implementing this. Probably i should be using the patch library or somthing? I'm new to python though so im not sure - but i didnt want to spend an age on it and this works. If this is wrong a point in the right direction would be handy. Perhaps longer term we would want to look at a full file system mock for testing (https://pypi.org/project/pyfakefs/) ?